dashboard-app/src/app/pages/facility/controllers/FacilityCtrl.js

```diff
@@ -70,5 +77,99 @@
             }
         };
 
+        $scope.openEditableFacilityForm = function () {
+            var selectedFacility = JSON.parse($window.localStorage.getItem('editableFacility'));
+            $scope.facility = selectedFacility;
+            $scope.loadImage(selectedFacility.image);
+        };
+
+        $scope.handleEditButton = function (selectedFacility) {
+            $window.localStorage.setItem('editableFacility', JSON.stringify(selectedFacility));
+            $state.go('facility');
+        };
+
+        $scope.handleAddFacilityButton = function () {
+            $state.go('facility');
+        };
+
+        $scope.closeFacilityModal = function () {
+            $uibModalStack.dismissAll();
+        };
+
+        $scope.handleDeleteButton = function (selectedFacility) {
+            $scope.selectedFacility = selectedFacility;
+            $uibModal.open({
+                templateUrl: 'app/pages/facility/views/facility-modal.html',
+                controller: 'FacilityCtrl',
+                scope: $scope
+            });
+        };
+
+        $scope.deleteFacility = function (selectedFacility) {
+
+            FacilityService.deleteFacility(selectedFacility.id).then(function (response) {
+
+                if (response.status === 200) {
+                    toastr.success(selectedFacility.facilityName + ' has been deleted successfully', 'Delete Facility');
+
+                    $scope.closeFacilityModal();
+                    $state.reload();
+                } else if (response.status === 500) {
+                    toastr.error('Facility ' + selectedFacility.facilityName + ' can not be deleted, it is already in use', 'Delete Facility');
+                } else {
```

Comment:
Checking for status 500 seems to be redundant, Facility is not FK so it can be deleted in any time